transcom/mymove
# MAIN B-21986

[INT PR 1](https://github.com/transcom/mymove/pull/14316)
[INT PR 2](https://github.com/transcom/mymove/pull/14317)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21986)

## Summary

We were calculating the authorized weight from the entitlements into the estimated incentive calculation for PPMs after sending some code to address the calculation for SSWs. Turns out these two calculations should NOT be using the same weight for the calculation.

### How to test

1. Access MM as customer
2. Create a PPM
3. Your estimated incentive shouldn't be crazy high

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21986)

## Summary

The saga continues.

With the change in the previous PR that fixed the estimated incentive to use the provided estimated weight, that changed the _previously_ implemented change to the SSW form to show the same value, which we do not want. The SSW show should show, under the max obligations, the price that reflects the MAX authorized weight and not the estimated weight, which can be wildly different.

A big update to this is changing the postal code source when pricing to use the orders locations and NOT the PPM shipment pickup/delivery addresses. This was approved by Dan Schuster.

Since we couldn't seem to share these two functionalities, I decided to store the value in the database and use it in the SSW. This will ensure we have different values between the estimated and max allowable.

Once the PPM reaches the `WAITING_ON_CUSTOMER` status, the max incentive should not change.

This PR also addresses updating and showing the PPM statuses throughout the workflow to office users.
<img width="787" alt="Screenshot 2024-12-03 at 11 20 33 AM" src="https://github.com/user-attachments/assets/ba9a2fa4-29a8-4ff8-a0fa-3578cca2b035">
<img width="795" alt="Screenshot 2024-12-03 at 11 20 51 AM" src="https://github.com/user-attachments/assets/4d89178f-d956-4ae8-b56b-a53b188871ed">
<img width="781" alt="Screenshot 2024-12-03 at 11 21 06 AM" src="https://github.com/user-attachments/assets/9f5f5123-bb7f-4193-856a-68db9626d7e2">


### How to test HACKY

1. Access MM as a customer
2. Create a PPM move with some light weight
3. Once you see the estimated incentive value 
5. Go check the db and you should see a different value in the `max_incentive` column
6. Click back
7. Put in the same weight as your allowance into the estimated weight of the PPM
8. Your estimated incentive and max incentive should now be the same

### How to test PROPERLY

1. Access MM as a customer
2. Create a PPM move
3. Once you see the estimated incentive value 
4. Go check the db and you should see a different value in the `max_incentive` column
5. Finish out your PPM, including closeout
9. Download the payment packet
10. Ensure that the price/value in the db shows under the `Maximum Obligations` section
11. Throughout this process, ensure you are seeing the updated tags for the office user when viewing the shipments

## Screenshots
After filling out PPM info, you can see the estimated incentive
<img width="638" alt="Screenshot 2024-12-02 at 10 50 12 PM" src="https://github.com/user-attachments/assets/6865fe04-a55a-42bc-be7c-8213f90e8d5d">
`estimated_incentive` in the db
<img width="258" alt="Screenshot 2024-12-02 at 10 50 37 PM" src="https://github.com/user-attachments/assets/46b4237e-b974-42c1-bd6b-b16c99055dc1">

On the SSW
<img width="642" alt="Screenshot 2024-12-02 at 10 52 57 PM" src="https://github.com/user-attachments/assets/13656913-dda2-4248-a98e-e11fa38bdda0">
`max_incentive` in the db
<img width="152" alt="Screenshot 2024-12-02 at 10 53 12 PM" src="https://github.com/user-attachments/assets/33c1198e-6e66-4a9e-b3ad-e203abc37bc1">
